insights/parsers/rhsm_releasever.py

```diff
@@ -43,11 +43,13 @@ def parse_content(self, content):
         """
         super(RhsmReleaseVer, self).parse_content(content)
         self.set = self.major = self.minor = None
+        rel_splits = []
         if 'releaseVer' not in self.data:
             raise SkipException('releaseVer is not in data')
 
         rel = self.data['releaseVer']
-        rel_splits = rel.split('.')
+        if rel is not None:
+            rel_splits = rel.split('.')
```

Comment:
@vishwanathjadhav - Better handle it with "no releaseVer" in the same way:

```diff
--- a/insights/parsers/rhsm_releasever.py
+++ b/insights/parsers/rhsm_releasever.py
@@ -43,13 +43,12 @@ class RhsmReleaseVer(JSONParser):
         """
         super(RhsmReleaseVer, self).parse_content(content)
         self.set = self.major = self.minor = None
-        rel_splits = []
-        if 'releaseVer' not in self.data:
-            raise SkipException('releaseVer is not in data')

-        rel = self.data['releaseVer']
-        if rel is not None:
-            rel_splits = rel.split('.')
+        rel = self.data.get('releaseVer')
+        if rel is None:
+            raise SkipException('releaseVer is not in data or not set')
+
+        rel_splits = rel.split('.')
```

Reply:
@vishwanathjadhav - you are right, we should not skip when there is 'releaseVer', but it's `null`

How about

```diff
--- a/insights/parsers/rhsm_releasever.py
+++ b/insights/parsers/rhsm_releasever.py
@@ -47,9 +47,8 @@ class RhsmReleaseVer(JSONParser):
         if 'releaseVer' not in self.data:
             raise SkipException('releaseVer is not in data')

-        rel = self.data['releaseVer']
-        if rel is not None:
-            rel_splits = rel.split('.')
+        rel = self.data.get('releaseVer') or ''
+        rel_splits = rel.split('.')
         # Release: 6.7
         if len(rel_splits) == 2:
             if rel_splits[0].isdigit() and rel_splits[-1].isdigit():
```

Reply:
Done